.github/workflows/build-python.yml

```diff
@@ -162,6 +168,14 @@ jobs:
         run: |
           bash .github/scripts/python.sh -d
 
+      - name: Create virtual on MacOS
```

Comment:
Any specific reason we want to use venvs instead of the installed Python? We could also extend the above `Setup python` step to install the desired python version.

PS I am not sure how python dev libraries play in a venv.

Reply:
Yes, very specific: on M1 the installed python is tagged as not modifiable. If you see the action failures and the many commits you'll see how I had to fight to get this working. There is no "just do this" solution that I could find. The added benefit is that now people will be able to use virtualenvs, whereas before they could not.

Reply:
> on M1 the installed python is tagged as not modifiable. 

The same goes on with Ubuntu >=24.04, by the way...